com.unity.formats.fbx/Editor/FbxExporter.cs

```diff
@@ -4628,7 +4628,7 @@ private static string ConvertToMayaCompatibleName(string name)
         {
             string newName = RemoveDiacritics (name);
 
-            if (char.IsDigit (newName [0])) {
+            if (newName.Length > 0 && char.IsDigit (newName [0])) {
```

Comment:
this could be string.IsNullOrEmpty() in case of null string (not sure if that's possible though)

Reply:
also will this leave the name empty? wondering if we should treat empty as an invalid char and replace it with `InvalidCharReplacement` as well, what do you think?

Reply:
The bug was that if the name was empty, you would get the error while trying to rename the fbx file in the export window. It wouldn't let you rename the fbx file, and you would have to cancel the export. 
This leaves the name empty, but lets you rename the file in the export window. The exporter makes you choose a filename for export anyways, since I don't think a ".fbx" file would work anywhere.

Reply:
I thought this would give the error even in a hierarchy where the root object exported has a name but the child does not. e.g.:

>GameObject
>> {empty name}

Reply:
It doesn't give an error, but the model inside the fbx object has no name. It doesn't really seem to effect anything; I was able to import the fbx into blender fine (I don't have a copy of Maya).

![empty](https://user-images.githubusercontent.com/85130337/120367136-8122c580-c2de-11eb-9805-cfb4f58e1a48.png)

I could assign the replacement character as a default name for that case though.